guide/src/building_and_distribution.md

```diff
@@ -1,58 +1,91 @@
+
+The material in this chapter is intended for users who have already read the PyO3 [README](#index.md). It covers in turn the choices that can be made for Python modules and for Rust binaries. There is also a section on the end about cross-compiling projects using PyO3.
+
+There is an additional sub-chapter dedicated to [supporting multiple Python versions](./building_and_distribution/multiple_python_versions.html).
+
+## Configuring the Python version
+
+PyO3 uses a build script (backed by the [`pyo3-build-config`] crate) to determine the Python version and set the correct linker arguments. By default it will attempt to use the following in order:
+ - Any active Python virtualenv.
+ - The `python` executable (if it's a Python 3 interpreter).
+ - The `python3` executable.
+
+You can override the Python interpreter by setting the `PYO3_PYTHON` environment variable, e.g. `PYO3_PYTHON=python3.6`, `PYO3_PYTHON=/usr/bin/python3.9`, or even a PyPy interpreter `PYO3_PYTHON=pypy3`.
+
+Once the Python interpreter is located, `pyo3-build-config` executes it to query the information in the `sysconfig` module which is needed to configure the rest of the compilation.
+
+To validate the configuration which PyO3 will use, you can run the `print-config` binary from the `pyo3-build-config` crate. An example usage of this binary is shown below:
+
+```bash
+# (First, clone the PyO3 git repository)
+$ cd path/to/pyo3/repository
+$ cd pyo3-build-config
+$ cargo run --bin print-config
+cargo:rerun-if-env-changed=PYO3_PYTHON
+cargo:rerun-if-env-changed=VIRTUAL_ENV
+cargo:rerun-if-env-changed=CONDA_PREFIX
+cargo:rerun-if-env-changed=PATH
+implementation: CPython
+interpreter version: 3.8
+interpreter path: Some("/usr/bin/python")
+libdir: Some("/usr/lib")
+shared: true
+base prefix: Some("/usr")
+ld_version: Some("3.8")
+pointer width: Some(8)
+```
 
-PyO3 uses a build script to determine the Python version and set the correct linker arguments. By default it uses the `python3` executable. You can override the Python interpreter by setting `PYO3_PYTHON`, e.g., `PYO3_PYTHON=python3.6`.
+## Building Python extension modules
 
-## Linking
+Python extension modules need to be compiled differently depending on the OS (and architecture) that they are being compiled for. As well as multiple OSes (and architectures). There are also many different Python versions which are actively supported  Packages uploaded to [PyPI](https://pypi.org/) usually want to upload prebuilt "wheels" covering many OS/arch/version combinations so that users on all these different platforms don't have to compile the package themselves. Package vendors can opt-in to the "abi3" limited Python API which allows their wheels to be used on multiple Python versions but restricts the functionality they can use, which reduces the number of wheels they need to compile.
```

Comment:
```suggestion
Python extension modules need to be compiled differently depending on the OS (and architecture) that they are being compiled for. As well as multiple OSes (and architectures), there are also many different Python versions which are actively supported. Packages uploaded to [PyPI](https://pypi.org/) usually want to upload prebuilt "wheels" covering many OS/arch/version combinations so that users on all these different platforms don't have to compile the package themselves. Package vendors can opt-in to the "abi3" limited Python API which allows their wheels to be used on multiple Python versions, reducing the number of wheels they need to compile, but restricts the functionality they can use.
```

Reply:
👍 thanks for catching so many typos in this paragraph!